### PR TITLE
Unit tests store api

### LINKED ISSUE #144: Write unit test for the API Vuex Store
**Why**
Increase our test coverage and give us increased confidence in the api store

**What**
* [ ] Test for state
* [ ] Test getters
* [ ] Test mutations
* [ ]  Test actions
* [ ] Add type checks to the store for backend choices 

**Notes**
Sub issue of #101

### PR DESCRIPTION
**Pull request by zhen0**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
- Adds unit tests for the api Vuex store
- Closes #144

### CODE REVIEW

File: tests/unit/store/api.spec.js [hunk 1]
```diff
@@ -0,0 +1,635 @@
...
+        // that we don't want to
+        // test here
+        api.mutations['tenant/setTenant'] = jest.fn()
+        api.mutations['tenant/setDefaultTenant'] = jest.fn()
+        api.mutations['tenant/unsetTenants'] = jest.fn()
+        api.mutations['tenant/unsetTenant'] = jest.fn()
+        api.actions['tenant/getTenants'] = jest.fn()
+        api.actions['auth0/authenticate'] = jest.fn()
+        api.actions['auth0/authorize'] = jest.fn()
+        api.actions['user/getUser'] = jest.fn()
```
Comment (znicholasbrown) [inline tests/unit/store/api.spec.js:564]: 👍

File: tests/unit/store/api.spec.js [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+      it('should set the connection message', async () => {
+        await store.dispatch('monitorConnection')
+        expect(store.getters.connectionMessage).toBe('Connected')
+      })
+      it('should set retries back to 0', async () => {
+        await store.dispatch('monitorConnection')
+        expect(store.getters.retries).toBe(0)
+      })
+    })
+    //Mock won't work as error policy set to 'none'
```
Comment (znicholasbrown) [inline tests/unit/store/api.spec.js:509]: If we can't test this part of the store, should we change the error policy on the `fallbackApolloClient` to let us catch issues? Doesn't necessarily need to be in this PR but seems like something we should be able to test.

File: tests/unit/store/api.spec.js [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+    })
+    it('should return the active url', () => {
+      expect(store.getters.url).toBe(undefined)
+    })
+    it('should return the version', () => {
+      expect(store.getters.version).toBe(null)
+    })
+  })
+
+  describe('where relevant getters check local storage before environment variables', () => {
```
Comment (znicholasbrown) [inline tests/unit/store/api.spec.js:189]: I don't think this is a true statement - none of the getters for the API store check `localStorage` or environment variables; only the initial state is set based on those. As such, I think just the above `State`-testing describe block will cover this, meaning we can combine all 3 getters describe blocks into 1 general use pattern.

File: tests/unit/store/api.spec.js (hunk 2 above)
Comment (zhen0) [reply tests/unit/store/api.spec.js]: Actually, I updated the mock and can now test this.  We also added a catch for if no data returned.  Tested locally and can't see any error.  (Not actually sure where we use that connectionMessage?)

**Comment by zhen0** [comment]
Ready for another review @znicholasbrown

**Pull request merged**